app/Http/Controllers/ViewController.php

```diff
@@ -53,8 +54,71 @@ public function show(Request $request, string $client, string $fileName): Respon
             $file = $cached;
         }
 
-        return response($file->binary())
-            ->header('Content-Type', $file->mimeType())
-            ->header('Cache-Control', 'public, max-age=' . env('CACHE_TIME', 15552000)); // default 6 months
+        return $file->isStreamable() ? $this->stream($request, $file)
+            : response($file->binary())
+                ->header('Content-Type', $file->mimeType())
+                ->header('Cache-Control', 'public, max-age=' . env('CACHE_TIME', 15552000)); // default 6 months
+    }
+
+    private function stream(Request $request, File $file): Response | StreamedResponse
+    {
+        ob_get_clean();
+        $buffer = 102400;
+        $streamPointer = $file->stream();
+        $start = 0;
+        $size = filesize($file->absolutePath());
+        $end = $size - 1;
+        $dynamicHeaders = [];
+
+        if ($request->server('HTTP_RANGE')) {
+            $request_start = $start;
+            $request_end = $end;
+
+            list(, $range) = explode('=', $request->server('HTTP_RANGE'), 2);
+
+            if ($range === '-') {
+                $request_start = $size - substr($range, 1);
+            } else {
+                $range = explode('-', $range);
+                $request_start = $range[0];
+
+                $request_end = (isset($range[1]) && is_numeric($range[1])) ? $range[1] : $request_end;
+            }
+            $request_end = ($request_end > $end) ? $end : $request_end;
+
+            $start = $request_start;
+            $end = $request_end;
+
+            fseek($streamPointer, $start);
+            $dynamicHeaders['Content-Length'] = $end - $start + 1;
+            $dynamicHeaders['Content-Range'] = "bytes $start-$end/$size";
+        } else {
+            $dynamicHeaders['Content-Length'] = $size;
+        }
+
+        return response()->stream(
+            function () use ($streamPointer, $buffer, $start, $size, $end) {
+                $i = $start;
```

Comment:
Zmienił bym i na coś bardziej informatywnego. Np obecny bajt. Ale nie można tego wyciągnąć czasem ze stream pointera? Bo jeśli nie to równie dobrze mogło by go nie być i byśmy używali tej zmiennej